client/gateway/spi/src/test/java/org/eclipse/kapua/client/gateway/spi/DefaultClientTest.java

```diff
@@ -0,0 +1,299 @@
+
+import org.eclipse.kapua.client.gateway.ErrorHandler;
+import org.eclipse.kapua.client.gateway.MessageHandler;
+import org.eclipse.kapua.client.gateway.Payload;
+import org.eclipse.kapua.client.gateway.Topic;
+import org.eclipse.kapua.qa.markers.junit.JUnitTests;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.mockito.Mockito;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+@Category(JUnitTests.class)
+public class DefaultClientTest extends Assert {
+
+    @Test
+    public void builderChannelIdTest() {
+        Channel channel = Mockito.mock(Channel.class);
+        DefaultClient.Builder builder = new DefaultClient.Builder(channel);
+        assertEquals("Expected and actual values should be the same.", builder, builder.module(Mockito.mock(Module.class)));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void builderChannelIdNullTest() {
+        DefaultClient.Builder builder = new DefaultClient.Builder(null);
+        assertEquals("Expected and actual values should be the same.", builder, builder.module(Mockito.mock(Module.class)));
+    }
+
+    @Test
+    public void builderTest() {
+        Channel channel = Mockito.mock(Channel.class);
+        DefaultClient.Builder builder = new DefaultClient.Builder(channel);
+        assertEquals("Expected and actual values should be the same.", builder, builder.builder());
+    }
+
+    @Test
+    public void buildTest() throws Exception {
+        Channel channel = Mockito.mock(Channel.class);
+        DefaultClient.Builder builder = new DefaultClient.Builder(channel);
+        DefaultClient defaultClientActual = (DefaultClient) builder.build();
+        final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+        final Set<Module> modules = new HashSet<>();
+        modules.add(Mockito.mock(Module.class));
+        DefaultClient client = new DefaultClient(channel, executorService, modules);
+        client.init();
+        assertEquals("Expected and actual values should be the same.", client.executor.isShutdown(), defaultClientActual.executor.isShutdown());
+    }
+
+    @Test
+    public void defaultClientTest() {
+        final Channel channel = Mockito.mock(Channel.class);
+        final ScheduledExecutorService executorService = Mockito.mock(ScheduledExecutorService.class);
+        final Set<Module> modules = new HashSet<>();
+        modules.add(Mockito.mock(Module.class));
+        DefaultClient client = new DefaultClient(channel, executorService, modules);
+        assertEquals("Expected and actual values should be the same.", executorService, client.executor);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void defaultClientChannelNullTest() {
```

Comment:
Please check this.
The test is either expected to throw NPE or perform the last `.assertEquals`.
In the first case `assertEquals` should be removed and let just the NPE happen